keps/sig-storage/20190129-csi-skip-attach.md

```diff
@@ -66,3 +66,5 @@ This KEP contains details that are missing from the design proposal.
 
 ## Implementation History
 * K8s 1.12: Alpha implementation
+* K8s 1.14: Beta implementation
+* K8s 1.17: GA implementation
```

Comment:
Have we thought about how an upgrade sequence would work for drivers?  How can a driver upgrade itself to use skip attach feature while there could be pending attaches in flight?  Is the implementation able to reconcile this and recover?

Reply:
There is an upgrade section in linked doc - https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface-skip-attach.md#upgrade . But basically, since code that checks for pending attachments also is designed to skip the check if volume does not require attach/detach.

I.e:
1. Attach initiated for a pod/volume that was previously attachable.
2. `VolumeAttachment` is created but may not finish attaching
2. `CSIDriver` object created that marks volume as non-attachable.
3. On Kubelet side code that checks for attachment will now recognize the fact that, volume does not support attachment and hence it will be correctly mounted.
3. Detach or Unmount though will still delete the previously created `VolumeAttachment` object.

Reply:
Awesome thanks! Should we add this scenario as a possible integration test case?

Reply:
Yeah sure. I will ensure that a test captures this.

Reply:
For kep purposes, adding it to the list of test cases above is sufficient

Reply:
done